manual_block_election.py

```diff
@@ -0,0 +1,86 @@
+import argparse
+import binascii
+
+import _logger
+import common
+from block import block_open
+from confirm_req import get_confirm_hash_resp
+from peercrawler import get_peers_from_service
+from pynanocoin import livectx, get_connected_socket_endpoint, betactx, testctx, get_genesis_block
+from msg_handshake import node_handshake_id
+from exceptions import PyNanoCoinException
+from representative_mapping import representative_mapping
+from common import hexlify
+
+logger = _logger.get_logger()
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('-b', '--beta', action='store_true', default=False,
+                       help='use beta network')
+    group.add_argument('-t', '--test', action='store_true', default=False,
+                       help='use test network')
+    parser.add_argument('-H', '--hash', type=str,
+                        default=None,
+                        help='the hash pair (in the form hash:root)')
+    return parser.parse_args()
+
+
+def main():
+    # eg hash: 991CF190094C00F0B68E2E5F75F6BEE95A2E0BD93CEAA4A6734DB9F19B728948
+    # eg root: E89208DD038FBB269987689621D52292AE9C35941A7484756ECCED92A65093BA
+    args = parse_args()
+    ctx = livectx
+
+    if args.beta:
+        ctx = betactx
+    elif args.test:
+        ctx = testctx
+
+    peers = filter(lambda p: p.is_voting, get_peers_from_service(ctx))
+    if args.hash is not None:
+        block_hash = args.hash.split(':')
+        if len(block_hash) == 1:
+            pair = common.hash_pair(binascii.unhexlify(block_hash[0]), b'\x00' * 32)
+        else:
+            pair = common.hash_pair(binascii.unhexlify(block_hash[0]), binascii.unhexlify(block_hash[1]))
+    else:
+        genesis_block = get_genesis_block(ctx)
+        pair = common.hash_pair(genesis_block.hash(), genesis_block.root())
+
+    votes = []
+    peers_voted = []
+    voting_weights = []
+    rep_map = representative_mapping()
+    rep_map.load_from_file("representative-mappings.json")
```

Comment:
Now that we have a representatives service, we should use it instead of this, I think, as this data is not live.